src/System Application/App/User Settings/src/UserSettingsImpl.Codeunit.al

```diff
@@ -368,6 +370,40 @@ codeunit 9175 "User Settings Impl."
         ApplicationUserSettings.Modify();
     end;
 
+    procedure LegacyActionBarEnabled(UserSecurityId: Guid): Boolean
+    var
+        ApplicationUserSettings: Record "Application User Settings";
+    begin
+        GetAppSettings(UserSecurityId, ApplicationUserSettings);
+        exit(ApplicationUserSettings."Legacy Action Bar");
+    end;
+
+    procedure DisableLegacyActionBar(UserSecurityId: Guid)
+    var
+        ApplicationUserSettings: Record "Application User Settings";
+    begin
+        GetAppSettings(UserSecurityId, ApplicationUserSettings);
+        ApplicationUserSettings."Legacy Action Bar" := false;
+        ApplicationUserSettings.Modify();
+    end;
+
+    procedure EnableLegacyActionBar(UserSecurityId: Guid)
+    var
+        ApplicationUserSettings: Record "Application User Settings";
+        CustomerExperienceSurvey: Codeunit "Customer Experience Survey";
+        FormsProId: Text;
+        FormsProEligibilityId: Text;
+        IsEligible: Boolean;
+    begin
+        GetAppSettings(UserSecurityId, ApplicationUserSettings);
+        ApplicationUserSettings."Legacy Action Bar" := true;
```

Comment:
We don't want to validate? That would allow partners to change this value with events (OnBeforeValidateField())

Reply:
or we want them to hook in on global modify trigger?

Reply:
I don't call validate. So those events aren't called.
If the partner decides to go ahead and subscribe fx "OnBeforeInsert", "OnAfterInsert" and make it false, that'd be their decision.